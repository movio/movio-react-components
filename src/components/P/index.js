import React, { PropTypes } from 'react';
import classnames from 'classnames';

import styles from './p.css';

const P = ({ className, children }) => (
  <p className={classnames(className, styles.p)}>
    {children}
  </p>
);

P.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};

P.defaultProps = {
  className: null,
};

export default P;
export { styles };
