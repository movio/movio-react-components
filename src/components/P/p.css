@import "variables.css";

:root {
  --p-color: var(--color-neutral-3);
  --p-font-size: 1rem;
  --p-font-family: var(--font-primary);
}

.p {
  font-family: var(--p-font-family);
  font-size: var(--p-font-size);
  line-height: 1.5;
  margin: 0;
  padding: 0;
  color: var(--p-color);
}

