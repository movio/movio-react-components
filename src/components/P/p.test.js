import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import P, { styles } from './index';

describe('<P />', () => {
  it('should correctly render children, when text', () => {
    const component = <P>This is a sample paragraph. It has some level of text. There is a lot of it.</P>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('This is a sample paragraph. It has some level of text. There is a lot of it.');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('p-basic');
  });

  it('should append a className to the default, if provided', () => {
    const component = <P className="test-class">This is a sample paragraph. It has some level of text. There is a lot of it.</P>; //eslint-disable-line
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.p}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('p-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <P>This is a sample paragraph. It has some level of text. There is a lot of it.</P>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.p);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('p-default-classname');
  });
});

