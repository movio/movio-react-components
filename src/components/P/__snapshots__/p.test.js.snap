// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p-basic 1`] = `
<p
  className="p"
>
  This is a sample paragraph. It has some level of text. There is a lot of it.
</p>
`;

exports[`p-custom-classname 1`] = `
<p
  className="test-class p"
>
  This is a sample paragraph. It has some level of text. There is a lot of it.
</p>
`;

exports[`p-default-classname 1`] = `
<p
  className="p"
>
  This is a sample paragraph. It has some level of text. There is a lot of it.
</p>
`;
